torch/_decomp/decompositions.py

```diff
@@ -1111,7 +1111,8 @@ def addmm(self: Tensor, mat1: Tensor, mat2: Tensor, beta: int = 1, alpha: int =
     out = alpha * torch.mm(mat1, mat2)
     if beta == 0:
         return out
-    return beta * self + out
+    # The output of addmm is contiguous, so we use `out`, which is contigous, as the first argument for add
```

Comment:
man... this feels bad... 
add is not associative for stride behavior... 
I can also use .contiguous(), but it won't be as efficient as this implementation.

Reply:
You mean commutative?
And that is~expected in python I think, that's why you have radd haha

Reply:
(in general I think our stride behavior is not associative, but yes on a single op it's not commutative)

Reply:
This is sort of a scary thing to rely upon, because I think this is an implementation detail of add, and not guaranteed.

If we do want to rely on this behavior I guess that's OK, but the comment should be updated to explain:

- what is being commented on (what's unusual about the line below)
- the goal (what are we trying to achieve)
- why a straightforward or naive version of the line would not achieve the desired result
- why the alternative version does achieve the desired result
- what limitations does the approach have, what assumptions does it make, and why we may need to change the behavior in the future

Reply:
Updated.